packages/@aws-cdk/aws-apigateway/lib/access-log.ts

```diff
@@ -0,0 +1,558 @@
+    return '$context.requestTimeEpoch';
+  }
+
+  /**
+   * The identifier that API Gateway assigns to your resource.
+   */
+  public static contextResourceId() {
+    return '$context.resourceId';
+  }
+
+  /**
+   * The path to your resource.
+   * For example, for the non-proxy request URI of `https://{rest-api-id.execute-api.{region}.amazonaws.com/{stage}/root/child`,
+   * The $context.resourcePath value is `/root/child`.
+   * For more information: https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-create-api-step-by-step.html
+   */
+  public static contextResourcePath() {
+    return '$context.resourcePath';
+  }
+
+  /**
+   * The deployment stage of the API request (for example, `Beta` or `Prod`).
+   */
+  public static contextStage() {
+    return '$context.stage';
+  }
+
+  /**
+   * The response received from AWS WAF: `WAF_ALLOW` or `WAF_BLOCK`. Will not be set if the stage is not associated with a web ACL.
+   * For more information: https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-control-access-aws-waf.html
+   */
+  public static contextWafResponseCode() {
+    return '$context.wafResponseCode';
+  }
+
+  /**
+   * The complete ARN of the web ACL that is used to decide whether to allow or block the request.
+   * Will not be set if the stage is not associated with a web ACL.
+   * For more information: https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-control-access-aws-waf.html
+   */
+  public static contextWebaclArn() {
+    return '$context.webaclArn';
+  }
+
+  /**
+   * The trace ID for the X-Ray trace.
+   * For more information: https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-enabling-xray.html
+   */
+  public static contextXrayTraceId() {
+    return '$context.xrayTraceId';
+  }
+
+  /**
+   * The authorizer latency in ms.
+   */
+  public static contextAuthorizerIntegrationLatency() {
+    return '$context.authorizer.integrationLatency';
+  }
+
+  /**
+   * The integration latency in ms.
+   */
+  public static contextIntegrationLatency() {
+    return '$context.integrationLatency';
+  }
+
+  /**
+   * For Lambda proxy integration, this parameter represents the status code returned from AWS Lambda,
+   * not from the backend Lambda function.
+   */
+  public static contextIntegrationStatus() {
+    return '$context.integrationStatus.';
+  }
+
+  /**
+   * The response latency in ms.
+   */
+  public static contextResponseLatency() {
+    return '$context.responseLatency';
+  }
+
+  /**
+   * The response payload length.
+   */
+  public static contextResponseLength() {
+    return '$context.responseLength';
+  }
+
+  /**
+   * The method response status.
+   */
+  public static contextStatus() {
+    return '$context.status';
+  }
+}
+
+/**
+ * A API Gateway custom access log format.
+ */
+export interface AccessLogOptions {
+  /**
+   * A API Gateway custom access log format
+   */
+  readonly format: string;
+}
```

Comment:
I believe we can now merge this with `AccessLogFormat` and eliminate the need for this class.

Like so -

```ts
export class AccessLogFormat {
  public static clf(): AccessLogFormat {
    ...
  }

  public static jsonWithStandardFields(...) {
    ...
  }

  public static custom(format: string): AccessLogFormat {
  }

  public toString(): string {
    ...
  }
}
```

Now replace all instances of `AccessLogOptions` with `AccessLogFormat`, and replace all instances of `accessLogFormat.format` with `accessLogFormat.toString()`.